Kernel/CMakeLists.txt

```diff
@@ -349,40 +349,31 @@ set(SOURCES
     ${CRYPTO_SOURCES}
 )
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-warning-option -Wvla -Wnull-dereference")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-rtti -ffreestanding -fbuiltin")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mno-80387 -mno-mmx -mno-sse -mno-sse2")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-asynchronous-unwind-tables")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fstack-protector-strong")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-exceptions")
+add_compile_options(-Wno-unknown-warning-option -Wvla -Wnull-dereference)
+add_compile_options(-fno-rtti -ffreestanding -fbuiltin)
+add_compile_options(-mno-80387 -mno-mmx -mno-sse -mno-sse2)
+add_compile_options(-fno-asynchronous-unwind-tables)
+add_compile_options(-fstack-protector-strong)
+add_compile_options(-fno-exceptions)
 # FIXME: remove -nodefaultlibs after the next toolchain update
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -nodefaultlibs -nostdlib")
+add_compile_options(-nodefaultlibs -nostdlib)
 
-if (USE_CLANG_TOOLCHAIN)
-    add_compile_options(-Waddress-of-packed-member)
-endif()
+add_compile_options("$<$<CXX_COMPILER_ID:Clang>:-Waddress-of-packed-member>")
```

Comment:
Oh, I thought that using the generator expression would be preferred to using an if(). Is that not the case? I can change all these back to ifs, but based based on CMAKE_CXX_COMPILER_ID instead of the custom USE_CLANG_TOOLCHAIN variable then

Reply:
It's just not really necessary to use a generator expression when what you are querying is already known at configure time. Genexes can also become quite unwieldy quickly and `if()` is just more readable.